e2e/tests/components/dashboard/dashboard.e2e-spec.ts

```diff
@@ -68,6 +68,38 @@ describe('Dashboard Tests', () => {
         expect(await imageCompareFullPageScreen('dashboard-option-change')).toEqual(0);
     });
 
+    it('should emit layoutChange and update the layout whenever a widget is dragged', async() => {
+        // drag the widget right
+        await browser.actions().dragAndDrop(widget2, { x: 250, y: 0 }).perform();
+
+        expect(await page.getWidgetLocationValue(widget1, 'top')).toBe(0);
+        expect(await page.getWidgetLocationValue(widget1, 'left')).toBe(0);
+
+        expect(await page.getWidgetLocationValue(widget2, 'top')).toBe(440);
+        expect(await page.getWidgetLocationValue(widget2, 'left')).toBe(277);
+
+        expect(await page.getWidgetLocationValue(widget3, 'top')).toBe(440);
+        expect(await page.getWidgetLocationValue(widget3, 'left')).toBe(0);
+
+        expect(await page.getWidgetLocationValue(widget4, 'top')).toBe(440);
+        expect(await page.getWidgetLocationValue(widget4, 'left')).toBe(831);
+
+       await page.setLayout.click();

```

Comment:
The karma test was covering the fact that the layout updates initially and when a new value is passed, this test is covering that the layout change is being emitted when drag because we couldn't implement the drag functionality easily  on a karma test

Reply:
I'm talking about this click and everything after. That part doesn't involve dragging and is covered by your karma test.